### ISSUE TITLE
Machines: can't limiting DHCP range for a virtual network, so using clients with static networking is risky

### ISSUE THREAD

**Issue by andreasn** [cockpit-project/cockpit#11259]
Page: Machines > Network

**Problem**
As a sysadmin, I want to have some VMs use static network addresses, so I need to limit the DHCP range on that subnet.

**User story**
Jeff is a new Linux admin. He's done a lot of system administration in the past, but on other OSes.
He runs a couple of VMs on a virt host that he manages using cockpit. 
He has a VM that he wants to keep using the same addresses all the time, so that all the other VMs can reach it in a predictable manner.

He logs in to the web console, dives into the VM and sets it's address to the static address 192.168.122.254.
He then edits it's virtual network to only give out new addresses and changes the DHCP range from 192.168.122.2-192.168.122.154 to 192.168.122.2-192.168.122.154
Now the DHCP server for the virtual network won't be able to accidentally hand out the 192.168.122.154 address.

**Comment by andreasn** [comment]
@skobyda filled out the user story as promised the other day.
Does that look good to you?

**Comment by skobyda** [comment]
Yes looks good, I just forgot to mention one thing.
With DHCP you can assign static IP address to a specific network interface. So "assigning static IP do VM is kinda indirectly true, because network interface is this kind of middle man between VM and virtual network. But every network interface can be assigned only to 1 VM.

So, when your sysadmin wants to assigning static IP address to specific network interface, user has to define 3 properties:
**Name**: specifies name of virtual dns server.
**MAC** (for IPv4) or **ID** (for IPv6): Is used for identification of network interface.
**IP**: the static address which is assigned to that interface.

**Comment by martinpitt** [comment]
@skobyda : This still seems current, right? I don't remember us working on a feature to set the DHCP range.

**Comment by skobyda** [comment]
Yes right this is still not implemented. Adding it to my todo-list
